src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -27,4 +27,31 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public int hashCode() {
+        return 31 * getLength() * getWidth()
+            * (getColor() != null ? getColor().length() : 2);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+
+        if (obj instanceof Rectangle) {
```

Comment:
what is the difference between method `instanceof` and `getClass()`?

Reply:
Oh, I understood it. I'll change it.